File: 2015/Day1/src/day1_ai.py
# Read input from file
try:
    with open('input.txt', 'r') as file:
        instructions = file.readline().strip()
except FileNotFoundError:
    print("Error: Unable to open input file")
    exit(1)

# Part 1: Calculate final floor
floor = 0
for char in instructions:
    if char == '(':
        floor += 1
    elif char == ')':
        floor -= 1

# Part 2: Find position of first basement entry
current_floor = 0
position = 0
for i, char in enumerate(instructions, 1):  # 1-based indexing
    if char == '(':
        current_floor += 1
    elif char == ')':
        current_floor -= 1
    if current_floor == -1:
        position = i
        break

# Output results
print(f"Part 1: Final floor = {floor}")
print(f"Part 2: First basement position = {position}")

File: 2015/Day1/src/day1_ai.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    // Read input from file
    std::ifstream file("input.txt");
    std::string instructions;
    if (file.is_open()) {
        std::getline(file, instructions);
        file.close();
    } else {
        std::cerr << "Unable to open input file\n";
        return 1;
    }

    // Part 1: Calculate final floor
    int floor = 0;
    for (char c : instructions) {
        if (c == '(') {
            floor++;
        } else if (c == ')') {
            floor--;
        }
    }

    // Part 2: Find position of first basement entry
    int position = 0;
    int current_floor = 0;
    for (size_t i = 0; i < instructions.length(); ++i) {
        if (instructions[i] == '(') {
            current_floor++;
        } else if (instructions[i] == ')') {
            current_floor--;
        }
        if (current_floor == -1) {
            position = i + 1; // 1-based indexing
            break;
        }
    }

    // Output results
    std::cout << "Part 1: Final floor = " << floor << "\n";
    std::cout << "Part 2: First basement position = " << position << "\n";

    return 0;
}
